src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -403,17 +293,20 @@ public ValueTask<bool> HasChangesAsync(Solution solution, SolutionActiveStatemen
 
             // Note that we may return empty deltas if all updates have been deferred.
             // The debugger will still call commit or discard on the update batch.
-            return new EmitSolutionUpdateResults(solutionUpdate.ModuleUpdates, solutionUpdate.Diagnostics);
+            return new EmitSolutionUpdateResults(solutionUpdate.ModuleUpdates, solutionUpdate.Diagnostics, solutionUpdate.DocumentsWithRudeEdits);
         }
 
-        public void CommitSolutionUpdate()
+        public void CommitSolutionUpdate(out ImmutableArray<DocumentId> documentsToReanalyze)
         {
             var editSession = _editSession;
             Contract.ThrowIfNull(editSession);
 
             var pendingUpdate = editSession.RetrievePendingUpdate();
             editSession.DebuggingSession.CommitSolutionUpdate(pendingUpdate);
-            editSession.ChangesApplied();
+
+            // restart edit session with no active statements (switching to run mode):
+            EndEditSession(out documentsToReanalyze);
+            StartEditSession(inBreakState: false);
```

Comment:
Do you need to do the same thing when discarding a solution update?

Reply:
No, because there is no transition to run mode when discarding update.